src/base/bittorrent/torrentimpl.cpp

```diff
@@ -2232,6 +2234,31 @@ QString TorrentImpl::createMagnetURI() const
     return QString::fromStdString(lt::make_magnet_uri(m_nativeHandle));
 }
 
+nonstd::expected<void, QString> TorrentImpl::exportToFile(const Path &path) const
+{
+    if (!hasMetadata())
+        return nonstd::make_unexpected(tr("Missing metadata"));
+
+    try
+    {
+        auto creator = lt::create_torrent(*(info().nativeInfo()));
+
+        for (const TrackerEntry &entry : asConst(trackers()))
+            creator.add_tracker(entry.url.toStdString(), entry.tier);
+
+        const lt::entry torrentEntry = creator.generate();
+        const nonstd::expected<void, QString> result = Utils::IO::saveToFile(path, torrentEntry);
```

Comment:
I really like this .torrent export functionality and hope to bring it to the webui. It would be helpful if this function dealt in bytes, or some other format, and left disk IO to the gui class (`TransferListWidget`). The webui could then send that over http, and the gui class could write it to the user supplied path.

Reply:
Do you mean export .torrent file to the client side?

Reply:
Sorry, I don't quite understand your question. Would you mind rephrasing?

Reply:
>I really like this .torrent export functionality and hope to bring it to the webui. 

I suppose you meant webapi?
I can do the changes in the torrent handle class and leave the webapi and webui parts to you or others, what do you say?

Reply:
That sounds great, thank you.

Reply:
>I can do the changes in the torrent handle class and leave the webapi and webui parts to you or others, what do you say?

Forgot to mention, that will be done in a later follow-up PR.